packages/swapper/src/swappers/RelaySwapper/endpoints.ts

```diff
@@ -217,6 +220,83 @@ export const relayApi: SwapperApi = {
 
     return feeData.fast.txFee
   },
+  getSolanaTransactionFees: async ({
+    tradeQuote,
+    from,
+    assertGetSolanaChainAdapter,
+  }: GetUnsignedSolanaTransactionArgs): Promise<string> => {
+    if (!isExecutableTradeQuote(tradeQuote)) throw Error('Unable to execute trade')
+
+    const firstStep = tradeQuote.steps[0]
+
+    const adapter = assertGetSolanaChainAdapter(firstStep.sellAsset.chainId)
+
+    if (!isExecutableTradeStep(firstStep)) throw Error('Unable to execute step')
+
+    const getFeeDataInput: GetFeeDataInput<KnownChainIds.SolanaMainnet> = {
+      to: '',
+      value: '0',
+      chainSpecific: {
+        from,
+        addressLookupTableAccounts:
+          firstStep.solanaTransactionMetadata?.addressLookupTableAddresses,
+        instructions: firstStep.solanaTransactionMetadata?.instructions,
+      },
+    }
+
+    const feeData = await adapter.getFeeData(getFeeDataInput)
+
+    return feeData.fast.txFee
+  },
+  getUnsignedSolanaTransaction: async ({
```

Comment:
Looking sane as it's quite literally the same as Relay, but same comment re: making a util to avoid effective 1/1 duplication

<img width="1459" alt="image" src="https://github.com/user-attachments/assets/39077ebd-bfb4-498e-9051-f50b0524e8db" />

Reply:
That's a nice catch, agree, keeping things DRY in [da24f87](https://github.com/shapeshift/web/pull/9253/commits/da24f87f9ab751c8b45c87a3b50e173b4cf2735b)